### DISCUSSION TITLE
ticks

### DISCUSSION THREAD

**Discussion by techOptimizor** [Jeiwan/uniswapv3-book#14] [Milestone 1. First Swap]
I wish to understand more about uniswap v3 ticks, am really confused. does the ticks specifically for the y reserve.

**Comment by Jeiwan** [comment]
> does the ticks specifically for the y reserve.

The ticks is a way to convert high-precision prices to a lower-precision scale. It's a gas optimization technique that disallows liquidity providers to provide liquidity at an arbitrary price (Uniswap uses a fixed-point number with high precision for prices). Since, during swapping, liquidity positions are traversed, we don't want to check too many ticks to not have very gas consumption.
Simply put, we don't want the scale that's used to place liquidity positions to have very high precision.

If you make a list of things that are not clear from reading the book, I'll improve the explanation.
